pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -146,9 +146,20 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 	logger = logger.WithValues("username", userSignup.Spec.Username)
 
+	// ###################################################################################
+	//
+	// TODO Migration code - remove this after all UserSignup instances have been migrated
+	//
+	// ###################################################################################
+	if userSignup.Spec.Deactivated && !states.Deactivated(userSignup) {
+		states.SetDeactivated(userSignup, true)
+		if err := r.client.Update(context.TODO(), userSignup); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
Could you please add a test that verifies the migration?

Reply:
Test added for migration, and migration will now return and requeue.